cmd/security-proxy-setup/main.go

```diff
@@ -0,0 +1,141 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.1.0
+ *******************************************************************************/
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/startup"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/usage"
+	"github.com/edgexfoundry/edgex-go/internal/security/proxy"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+func main() {
+
+	if len(os.Args) < 2 {
+		usage.HelpCallbackSecurityProxy()
+	}
+	var initNeeded bool
+	var insecureSkipVerify bool
+	var resetNeeded bool
+	var useProfile string
+	var userTobeCreated string
+	var userofGroup string
+	var userTobeDeleted string
+	var useRegistry bool
+
+	flag.BoolVar(&insecureSkipVerify, "insureskipverify", true, "skip server side SSL verification, mainly for self-signed cert")
+	flag.BoolVar(&initNeeded, "init", false, "run init procedure for security service.")
+	flag.BoolVar(&resetNeeded, "reset", false, "reset reverse proxy by removing all services/routes/consumers")
+	flag.StringVar(&userTobeCreated, "useradd", "", "user that needs to be added to consume the edgex services")
+	flag.StringVar(&userofGroup, "group", "user", "group that the user belongs to. By default it is in user group")
+	flag.StringVar(&userTobeDeleted, "userdel", "", "user that needs to be deleted from the edgex services")
+	flag.BoolVar(&useRegistry, "registry", false, "Indicates the service should use registry service.")
+	flag.BoolVar(&useRegistry, "r", false, "Indicates the service should use registry service.")
+	flag.StringVar(&useProfile, "profile", "", "Specify a profile other than default.")
+	flag.StringVar(&useProfile, "p", "", "Specify a profile other than default.")
+
+	flag.Usage = usage.HelpCallbackSecurityProxy
+	flag.Parse()
+
+	params := startup.BootParams{UseRegistry: useRegistry, UseProfile: useProfile, BootTimeout: internal.BootTimeoutDefault}
+	startup.Bootstrap(params, proxy.Retry, logBeforeInit)
+
+	req := proxy.NewRequestor(insecureSkipVerify)
+	s := proxy.NewService(req)
+
+	err := s.CheckProxyServiceStatus()
+	if err != nil {
+		proxy.LoggingClient.Error(err.Error())
+		return
+	}
+
+	if initNeeded {
+		if resetNeeded {
+			proxy.LoggingClient.Error("can't run initialization and reset at the same time for security service")
+			return
+		}
+		cert := proxy.NewCertificateLoader(req, proxy.Configuration.SecretService.CertPath, proxy.Configuration.SecretService.TokenPath)
+		err = s.Init(cert)
+		if err != nil {
+			proxy.LoggingClient.Error(err.Error())
+		}
+	}
+
+	if resetNeeded {
```

Comment:
we can use `else if resetNeeded` here as it is mutually exclusive with `initNeeded`